contracts/interchain-token-service/src/contract.rs

```diff
@@ -186,52 +113,61 @@ impl InterchainTokenServiceInterface for InterchainTokenService {
             .into()
     }
 
-    fn deploy_remote_interchain_token(
-        env: &Env,
-        caller: Address,
-        destination_chain: String,
-        _token_id: String,
-        gas_token: Token,
-    ) {
-        let destination_address = String::from_str(env, "");
-
-        // TODO: abi encode with MessageType.DeployInterchainToken
-        let payload = bytes!(env,);
+    fn deploy_interchain_token(
+        _env: &Env,
+        _caller: Address,
+        _salt: BytesN<32>,
+        _destination_chain: String,
+        _token_metadata: TokenMetadata,
+        _minter: Option<Bytes>,
+        _gas_token: Token,
+    ) -> Result<BytesN<32>, ContractError> {
+        todo!()
+    }
 
-        Self::pay_gas_and_call_contract(
-            env.clone(),
-            caller,
-            destination_chain,
-            destination_address,
-            payload,
-            gas_token,
-        );
+    fn deploy_remote_interchain_token(
+        _env: &Env,
+        _caller: Address,
+        _salt: BytesN<32>,
+        _minter: Option<Bytes>,
+        _destination_chain: String,
+        _gas_token: Token,
+    ) -> Result<BytesN<32>, ContractError> {
+        // TODO: implementation
+        // deploy salt
+        // minter approval
+        // registered token address - get token metadata
+        // abi encode with MessageType.DeployInterchainToken
+        // token id
+        // pay_gas_and_call_contract
+        todo!()
     }
 
     fn interchain_transfer(
         env: &Env,
         caller: Address,
-        _token_id: String,
-        _source_address: Bytes,
+        token_id: BytesN<32>,
         destination_chain: String,
         destination_address: Bytes,
-        _amount: i128,
-        _metadata: Bytes,
+        amount: i128,
+        metadata: Option<Bytes>,
         gas_token: Token,
-    ) {
-        // TODO: _takeToken, decode metadata, and abi encode with MessageType.InterchainTransfer
-        let payload = bytes!(&env,);
+    ) -> Result<(), ContractError> {
+        caller.require_auth();
 
-        // TODO: Get the params for the cross-chain message, taking routing via ITS Hub into account.
+        // TODO: implementation
 
-        Self::pay_gas_and_call_contract(
-            env.clone(),
-            caller,
-            destination_chain,
-            String::from_val(env, &destination_address.to_val()),
-            payload,
-            gas_token,
-        );
+        let message = Message::InterchainTransfer(InterchainTransfer {
+            token_id,
+            source_address: caller.clone().to_xdr(env),
```

Comment:
How does Address when encoded in XDR look like? is it just the raw bytes, or is there other metadata. We would like to encode the raw address bytes, it's easier to debug and reason about then some custom data format, similar to EVM accounts